File: app/src/main/java/co/tpcreative/supersafe/common/request/TrackingRequest.java
package co.tpcreative.supersafe.common.request;
import java.io.Serializable;

import co.tpcreative.supersafe.BuildConfig;
import co.tpcreative.supersafe.R;
import co.tpcreative.supersafe.common.services.SuperSafeApplication;

public class TrackingRequest implements Serializable {
    public String user_id;
    public String device_id;
    public String device_type;
    public String manufacturer;
    public String name_model;
    public String version;
    public String versionRelease;
    public String appVersionRelease;
    public String app_id;
    public String channel_code;

    public TrackingRequest(String user_id,String device_id){
        this.user_id = user_id;
        this.device_id = device_id;
        this.device_type = SuperSafeApplication.getInstance().getString(R.string.device_type);
        this.manufacturer =  SuperSafeApplication.getInstance().getManufacturer();
        this.name_model = SuperSafeApplication.getInstance().getModel();
        this.version = ""+SuperSafeApplication.getInstance().getVersion();
        this.versionRelease = SuperSafeApplication.getInstance().getVersionRelease();
        this.appVersionRelease =  BuildConfig.VERSION_NAME;
        this.app_id = SuperSafeApplication.getInstance().getPackageId();
        this.channel_code = "C002";
    }
}

File: app/src/main/java/co/tpcreative/supersafe/common/request/TrackingRequest.kt
package co.tpcreative.supersafe.common.request
import co.tpcreative.supersafe.BuildConfig
import co.tpcreative.supersafe.R
import co.tpcreative.supersafe.common.services.SuperSafeApplication
import java.io.Serializable
class TrackingRequest(var user_id: String?, var device_id: String?) : Serializable {
    var device_type: String?
    var manufacturer: String?
    var name_model: String?
    var version: String?
    var versionRelease: String?
    var appVersionRelease: String?
    var app_id: String?
    var channel_code: String?

    init {
        device_type = SuperSafeApplication.getInstance().getString(R.string.device_type)
        manufacturer = SuperSafeApplication.getInstance().getManufacturer()
        name_model = SuperSafeApplication.getInstance().getModel()
        version = "" + SuperSafeApplication.getInstance().getVersion()
        versionRelease = SuperSafeApplication.getInstance().getVersionRelease()
        appVersionRelease = BuildConfig.VERSION_NAME
        app_id = SuperSafeApplication.getInstance().getPackageId()
        channel_code = "C002"
    }
}